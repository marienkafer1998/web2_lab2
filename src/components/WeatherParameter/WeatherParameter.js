import React from "react";
import "./WeatherParameter.css";


const WeatherParameter = props => (
    <div className="param">
        <div className="name">{props.name}</div>
        <div className="value">{props.value}</div>
    </div>

);

export default WeatherParameter;