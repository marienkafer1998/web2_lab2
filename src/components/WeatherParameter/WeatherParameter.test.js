import React from 'react';
import renderer from 'react-test-renderer';
import WeatherParameter from "./WeatherParameter.js";

describe('WeatherParameter functional rendered properly', () => {

    const paramExample = {
        name: "Temperature",
        value: "1.5",
    };

    test("should has name and value", () => {
        const {name, value} = paramExample;

        const tree = renderer
            .create(
                <WeatherParameter
                    name={name}
                    value={value}
                />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });


    test("should has only name", () => {
        const {name} = paramExample;

        const tree = renderer
            .create(
                <WeatherParameter
                    name={name}
                />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test("should has only value", () => {
        const {value} = paramExample;

        const tree = renderer
            .create(
                <WeatherParameter
                    value={value}
                />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test("should has nothing", () => {
        const tree = renderer
            .create(
                <WeatherParameter/>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});