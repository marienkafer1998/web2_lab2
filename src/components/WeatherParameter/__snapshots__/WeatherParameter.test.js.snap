// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherParameter functional rendered properly 1`] = `
<div
  className="param"
>
  <div
    className="name"
  />
  <div
    className="value"
  />
</div>
`;

exports[`WeatherParameter functional rendered properly should has name and value 1`] = `
<div
  className="param"
>
  <div
    className="name"
  >
    Temperature
  </div>
  <div
    className="value"
  >
    1.5
  </div>
</div>
`;

exports[`WeatherParameter functional rendered properly should has nothing 1`] = `
<div
  className="param"
>
  <div
    className="name"
  />
  <div
    className="value"
  />
</div>
`;

exports[`WeatherParameter functional rendered properly should has only name 1`] = `
<div
  className="param"
>
  <div
    className="name"
  >
    Temperature
  </div>
  <div
    className="value"
  />
</div>
`;

exports[`WeatherParameter functional rendered properly should has only value 1`] = `
<div
  className="param"
>
  <div
    className="name"
  />
  <div
    className="value"
  >
    1.5
  </div>
</div>
`;
